### ISSUE TITLE
WIP: snapclient 0.28.0 (armbian/armv7l) despite repeated attempts cannot connect to snapserver 0.28.0 (openwrt/arm64) after some time.

### ISSUE THREAD

**Issue by davidandreoletti** [badaix/snapcast#1243]
Note: The information needed to figure out what's happening is still being gathered. Relevant fields will be updated as discovery process progresses.

**Describe the bug**
- After a hardware reboot, both snapcast client and server are started, the client can connect to the server and receives music.
- After that and upon an unknown set of conditions, the snapcast client cannot connect to the server. Pinging the server machine works and netstats report the server is still listening on port 1704/1705/1780.

**Steps to Reproduce**
1. No confirmed step available at the moment
2. (suspected) Snapserver or snapclient idle for 2+ days
3.

** Issue Ruled out **
- Snapclient machine can ping the snapserver host IP successfully.
- Firewall port on server  machine open since snapclient can connect to the snapserver successfully.

**Environment details**
- OS: (server) OpenWRT SNAPSHOT (client) armbian 23.05
- Snapcast version: (server + client) 0.28.0

**Log files**

**Snapserver logs after --hardware-- reboot: client can connect to the server**
```log
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Version 0.28.0
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Adding source: pipe:///tmp/snapfifo?name=default
Mon Jun  3 14:59:56 2024 user.notice snapserver[10312]: Settings file: "//.config/snapserver/server.json"
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Adding service 'Snapcast'
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Using HTTP host name: davida-router-hw1
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: PcmStream: default, sampleFormat: 48000:16:2
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: PipeStream mode: create
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Stream: {"fragment":"","host":"","path":"/tmp/snapfifo","query":{"chunk_ms":"20","codec":"flac","name":"default","sampleformat":"48000:16:2"},"raw":"pipe:////tmp/snapfifo?chunk_ms=20&codec=flac&name=default&sampleformat=48000:16:2","scheme":"pipe"}
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Init - compression level: 2
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Creating TCP acceptor for address: 0.0.0.0, port: 1705
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Creating HTTP acceptor for address: 0.0.0.0, port: 1780
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Creating stream acceptor for address: 0.0.0.0, port: 1704
Mon Jun  3 14:59:56 2024 user.info snapserver[10312]: Number of threads: 4, hw threads: 6
Mon Jun  3 14:59:56 2024 user.err snapserver[10312]: Error reading message: End of file, length: 0, ec: asio.misc:2
Mon Jun  3 14:59:57 2024 user.info snapserver[10312]: Service 'Snapcast' successfully established.
Mon Jun  3 15:00:05 2024 user.notice snapserver[10312]: StreamServer::NewConnection: 192.168.3.137
Mon Jun  3 15:00:05 2024 user.info snapserver[10312]: Hello from davida-audiospeaker-hw0, host: davida-audiospeaker-hw0, v0.28.0, ClientName: Snapclient, OS: Armbian-unofficial 24.5.0-trunk bookworm, Arch: armv7l, Protocol version: 2
```

**Snapserver logs after --snapserver-- reboot and unknown set of conditions triggered: client cannot connect to the server**
```log
Not yet available
```

**Comment by gpayer** [comment]
I have kind of the same problem, not sure. As soon as the music is stopped no new chunks are recognized, it's always "waiting for new chunks". Only a restart gets the music playing.

**Comment by davidandreoletti** [comment]
The issue described is different:
- The snapcast server (machine A) is up
- The snapcast client (machine B) is up and connected to the snapcast server. I can play music
- Stop playing music for a few days
- Try playing music again ... no music played because the snapcast client can't connect to the snapcast server (still up / no network issue nor firewall problem).

**Comment by badaix** [comment]
> Note: The information needed to figure out what's happening is still being gathered. Relevant fields will be updated as discovery process progresses.

Please reopen, when you've gathered all data. Client logs would help. I've once seen a client failing to resolve the MDNS-broadcasted server IP, so it might help to pass the server's IP to the client with `--host`.

**Comment by davidandreoletti** [comment]
@badaix I have more info on this issue and only you can reopen the ticket. I will post the detail here for now and rearrange the new info in the ticket once reopened.


Snapclient logs when the reconnection fails:
```
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.111 [Info] (Connection) Resolving host IP for: snapserver.audio.mediacenter.home
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.126 [Debug] (Connection) Resolved IP: fd24:f3fd:b0e5::1
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.126 [Debug] (Connection) Resolved IP: 192.168.3.1
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.126 [Info] (Connection) Connecting to [fd24:f3fd:b0e5::1]:1704
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.130 [Info] (Connection) Connecting to 192.168.3.1:1704
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.133 [Error] (Connection) Failed to connect to host 'snapserver.audio.mediacenter.home', error: Connection refused
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.134 [Error] (Controller) Error: Connection refused
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.134 [Debug] (Connection) Disconnecting
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.134 [Error] (Connection) Error in socket shutdown: Transport endpoint is not connected
Aug 18 13:34:22 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-22.134 [Debug] (Connection) Disconnected
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.135 [Info] (Connection) Resolving host IP for: snapserver.audio.mediacenter.home
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.153 [Debug] (Connection) Resolved IP: fd24:f3fd:b0e5::1
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.153 [Debug] (Connection) Resolved IP: 192.168.3.1
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.153 [Info] (Connection) Connecting to [fd24:f3fd:b0e5::1]:1704
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.157 [Info] (Connection) Connecting to 192.168.3.1:1704
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.160 [Error] (Connection) Failed to connect to host 'snapserver.audio.mediacenter.home', error: Connection refused
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.160 [Error] (Controller) Error: Connection refused
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.161 [Debug] (Connection) Disconnecting
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.161 [Error] (Connection) Error in socket shutdown: Transport endpoint is not connected
Aug 18 13:34:23 davida-audiospeaker-hw0 snapclient[1410]: 2024-08-18 13-34-23.161 [Debug] (Connection) Disconnected
```

When the reconnection fails from the snapclient, the snapcast server is definitely reachable by other means
```
nc snapserver.audio.mediacenter.home 1704

...non intelligble output as expected...

+L%(lD*LZ)*E"<2PH(`d    Cf$A %7d&bE*
          %9+f-Jn*-Y*(=%XB"Qn
RUbEm.[|VB[QѺY
              8P
                @@f -h57b-JByB-a=B"&jK !Bdf$Bf&
                                               f=DU!
                                                    "'lBQ       y@YP$$px 3O,t$  JlyS)pAa!(L!lܤjB)|HG
                                                                                                    !#V!x
...
```
